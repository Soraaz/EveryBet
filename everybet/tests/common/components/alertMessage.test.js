import React from "react";
import renderer from "react-test-renderer";

import AlertMessage from "../../../src/common/components/alertMessage.js";

describe("AlertMessage", () => {
	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<AlertMessage />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly simple alert", () => {
			const tree = renderer
				.create(<AlertMessage
					type={'simple'}
					title={'testTitle'}
					message={'testMessage'} />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly error", () => {
			const tree = renderer
				.create(<AlertMessage
					type={'error'}
					callback={jest.fn()}
					message={'testMessage'} />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly success alert", () => {
			const tree = renderer
				.create(<AlertMessage
					type={'success'}
					title={"testTitle"}
					params={null}
					callback={jest.fn()}
					message={'testMessage'} />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly success two buttons alert", () => {
			const tree = renderer
				.create(<AlertMessage
					type={'successTwoButtons'}
					title={"testTitle"}
					params={null}
					callback={jest.fn()}
					callback2={jest.fn()}
					message={'testMessage'}
					buttonMessage={"buttonMessage"}
					buttonMessage2={"buttonMessage2"}
					param2={null} />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly success two buttons alert", () => {
			const tree = renderer
				.create(<AlertMessage
					type={'simpleError'}
					message={'testMessage'} />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("funcionalities", () => {
		let component;

		beforeEach(() => {
			component = new AlertMessage();
		});

		it("Should instantiate the class", () => {
			expect(component).to.be.an.instanceOf(AlertMessage);
		});
	});
});
