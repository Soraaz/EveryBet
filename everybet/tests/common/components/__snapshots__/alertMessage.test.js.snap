// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage rendering Should render correctly 1`] = `<View />`;

exports[`AlertMessage rendering Should render correctly error 1`] = `<View />`;

exports[`AlertMessage rendering Should render correctly simple alert 1`] = `<View />`;

exports[`AlertMessage rendering Should render correctly success alert 1`] = `<View />`;

exports[`AlertMessage rendering Should render correctly success two buttons alert 1`] = `<View />`;

exports[`AlertMessage rendering Should render correctly success two buttons alert 2`] = `<View />`;
